app/helpers/comments_helper.rb

```diff
@@ -0,0 +1,8 @@
+# frozen_string_literal: true
+
+module CommentsHelper
+  def user_can_comment?(idea, user)
+    return idea.approved_by_admin? if user.admin
+    return idea.approved_by_admin? && !idea.not_proceeding? unless user.admin
```

Comment:
I've add this helper so that it is clearer in the view and controller, that we are checking that the user can comment, and because the logic of admin / not admin / approved / not proceeding etc is better off in a separate method now it's more complex.  But I feel that the logic here now is hard to read, would it be better to refactor further into other methods in this class such as `can_normal_user_comment?` and `can_admin_user_comment?`

Reply:
Good question! This isn't the ideal place, but let's leave it for now. I understand @stephenrichards will cover Pundit policies on Tuesday, that's where you'd put code like this.

Reply:
Cool, I'll merge this for now then and create a new PR after I've learnt about Pundit policies 👍